python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -387,60 +381,46 @@ def compute_intersection_with_partition_keys(
-                    included_time_windows=[time_window],
-                ),
-            ),
-        )
-
-    def create_latest_time_window_slice(self, asset_key: AssetKey) -> AssetSlice:
-        """If the underlying asset is time-window partitioned, this will return the latest complete
-        time window relative to the effective date. For example if it is daily partitioned starting
-        at midnight every day.  If the effective date is before the start of the partition definition, this will
-        return the empty time window (where both start and end are datetime.max).
-
-        If the underlying asset is unpartitioned or static partitioned and it is not empty,
-        this will return a time window from the beginning of time to the effective date. If
-        it is empty it will return the empty time window.
-
-        TODO: add language for multi-dimensional partitioning when we support it
-        TODO: add language for dynamic partitioning when we support it
+    def compute_latest_time_window_slice(
+        self, asset_key: AssetKey, lookback_timedelta: Optional[timedelta] = None
+    ) -> AssetSlice:
+        """Compute the slice of the asset which exists within the latest time partition window. If
+        the asset has no time dimension, this will always return the full slice. If
+        lookback_timedelta is provided, all partitions that are up to that timedelta before the
+        end of the latest time partition window will be included.
         """
         partitions_def = self._get_partitions_def(asset_key)
-        if partitions_def is None or isinstance(
-            partitions_def, (DynamicPartitionsDefinition, StaticPartitionsDefinition)
-        ):
+        time_partitions_def = get_time_partitions_def(partitions_def)
+        if time_partitions_def is None:
+            # if the asset has no time dimension, then return a full slice
             return self.get_asset_slice(asset_key)
 
-        if isinstance(partitions_def, TimeWindowPartitionsDefinition):
-            time_window = partitions_def.get_last_partition_window(self.effective_dt)
-            return (
-                self.create_from_time_window(asset_key, time_window)
-                if time_window
-                else self.create_empty_slice(asset_key)
+        latest_time_window = time_partitions_def.get_last_partition_window(self.effective_dt)
+        if latest_time_window is None:
+            return self.create_empty_slice(asset_key)
+
+        # the time window in which to look for partitions
+        time_window = (
+            TimeWindow(
+                start=max(
+                    # do not look before the start of the definition
+                    time_partitions_def.start,
+                    latest_time_window.end - lookback_timedelta,
+                ),
+                end=latest_time_window.end,
             )
+            if lookback_timedelta
+            else latest_time_window
+        )
```

Comment:
suspect this is generally useful functionality so consider extracting into its own function

Reply:
keeping it inlined for now, as it's unclear as of now the precise semantics a generic function (i.e. do we want it to handle both unpartitioned and partitioned, or just time-partitioned, etc.) would have, and I'd like to have another use case before extracting

Reply:
cool